sdk/cognitivelanguage/Azure.AI.Language.QuestionAnswering/src/QuestionAnsweringClient.cs

```diff
@@ -0,0 +1,179 @@
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.AI.Language.QuestionAnswering
+{
+    /// <summary>
+    /// The <see cref="QuestionAnsweringClient"/> allows you to ask questions of a custom or built-in knowledgebase.
+    /// </summary>
+    public class QuestionAnsweringClient
+    {
+        internal const string AuthorizationHeader = "Ocp-Apim-Subscription-Key";
+
+        private readonly QuestionAnsweringKnowledgebaseRestClient _knowledgebaseRestClient;
+        private readonly QuestionAnsweringTextRestClient _textRestClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QuestionAnsweringClient"/> class.
+        /// </summary>
+        /// <param name="endpoint">The Question Answering endpoint on which to operate.</param>
+        /// <param name="credential">A <see cref="AzureKeyCredential"/> used to authenticate requests to the <paramref name="endpoint"/>.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="endpoint"/> or <paramref name="credential"/> is null.</exception>
+        public QuestionAnsweringClient(Uri endpoint, AzureKeyCredential credential) : this(endpoint, credential, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QuestionAnsweringClient"/> class.
+        /// </summary>
+        /// <param name="endpoint">The Question Answering endpoint on which to operate.</param>
+        /// <param name="credential">A <see cref="AzureKeyCredential"/> used to authenticate requests to the <paramref name="endpoint"/>.</param>
+        /// <param name="options">Optional <see cref="QuestionAnsweringClientOptions"/> to customize requests sent to the endpoint.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="endpoint"/> or <paramref name="credential"/> is null.</exception>
+        public QuestionAnsweringClient(Uri endpoint, AzureKeyCredential credential, QuestionAnsweringClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNull(credential, nameof(credential));
+
+            Endpoint = endpoint;
+            options ??= new QuestionAnsweringClientOptions();
+
+            Diagnostics = new ClientDiagnostics(options);
+            Pipeline = HttpPipelineBuilder.Build(
+                options,
+                new AzureKeyCredentialPolicy(credential, AuthorizationHeader));
+
+            _knowledgebaseRestClient = new(Diagnostics, Pipeline, Endpoint, options.Version);
+            _textRestClient = new(Diagnostics, Pipeline, Endpoint, options.Version);
+        }
+
+        /// <summary>
+        /// Protected constructor to allow mocking.
+        /// </summary>
+        protected QuestionAnsweringClient()
+        {
+        }
+
+        /// <summary>
+        /// Get the service endpoint for this client.
+        /// </summary>
+        public virtual Uri Endpoint { get; }
```

Comment:
ohh interesting. I didn't know we were exposing the `endpoint` in the client. is this a new guideline?

Reply:
We did for Key Vault and Search. The idea was that it's easy to tell from a client with which endpoint you're communicating. It's not a guideline as much as a convenience. Tagging @KrzysztofCwalina and @tg-msft in case they want me to remove this, but it's been through approval in those aforementioned libraries. In fact, @tg-msft was the one who added it to Search first.

Reply:
ohh cool, thanks!